Enable native backend for DragonFly BSD

mneumann opened the pull request:
Without this, it's not possible to build ocamlopt on DragonFly BSD.

Tested-on: DragonFly 6.5

mneumann committed:
Enable native backend for DragonFly BSD

Without this, it's not possible to build ocamlopt on DragonFly BSD.

Tested-on: DragonFly 6.5

gasche commented:
Have you tried running the testsuite on DragonFly BSD? (I wonder if native compilation was disabled because some things are not supported.)

kit-ty-kate commented:
I have 6.4.2 installed at my local hackerspace. I'm happy to try running the testsuite with this change for double-checking purpose.

> Tested-on: DragonFly 6.5

Since the latest release is 6.4.2 i'm guessing you mean the master branch?

mneumann commented:
@gasche Yes, I ran "gmake tests" with success after the change. I also can use ocamlopt to compile native code executables on my system after the change (before ocamlopt just is not built).

I think it just got forgotten in commit c40b14b46ba by Xavier. The commit mentions "For 5.00 it's only x86_64 on Linux, macOS, BSD, and Mingw, IIUC.". DragonFly BSD is a *BSD* deriving from FreeBSD 4, but it doesn't have "bsd" in "uname -s" for instance.

Just for reference, my `uname -a` and `ocamlopt -v` output is:

```
dev/ocaml % uname -a
DragonFly letsnote.localnet 6.5-DEVELOPMENT DragonFly v6.5.0.1202.gd5712-DEVELOPMENT #23: Mon Dec 22 21:54:04 CET 2025     mneumann@letsnote.localnet:/usr/obj/usr/src/sys/X86_64_GENERIC  x86_64


dev/ocaml % /build/dev/opt/ocaml/bin/ocamlopt -v
The OCaml native-code compiler, version 5.5.0+dev0-2025-04-28
Standard library directory: /build/dev/opt/ocaml/lib/ocaml
```

btw, thanks for making Ocaml so easy to compile!!!

mneumann commented:
@kit-ty-kate yes, 6.5 is the master branch and my version is a few days old :-). Feel free to try on 6.4.2!!!

gasche commented:
Ah, that makes sense (that it fell off the radar due to not having `bsd`), thanks for digging the explanation up.

@kit-ty-kate I'm happy to approve and merge now, is there something else you want to do before that?

kit-ty-kate reviewed (approved):
I just ran the testsuite with this patch on 6.4.2 and it ran just fine at the exception of `tests/lib-runtime-events/test_create_cursor_failures.ml` due to the following line: https://github.com/ocaml/ocaml/blob/2922879136e8af6d20b930ba969c34ac7f871773/testsuite/tests/lib-runtime-events/test_create_cursor_failures.ml#L36

Testing using the `chmod(1)` tool, i get the following behaviour:
```
$ echo content > test
$ chmod 000 test
$ cat test
content
```

However this is due to the fact that i was testing using the `root` user, and `root` can read any file regardless of their permissions. (https://github.com/ocaml/ocaml/pull/14138 also hits this edge-case so i think it would be nice to have a guard for it)

Anyway, all that to say that i've tested both `make tests` and `make -C testsuite parallel` with a fresh user and they work just fine. I've also compiled and ran `opam` without any issue either.

@gasche feel free to merge modulo the missing entry in the changelog and the addition to the supported platform table in the README (Tier 2, x86_64)

mneumann committed:
Add Changelog entry

mneumann committed:
Mention DragonFly has Tier 2 support of native-code compiler

I like mentioning DragonFly in close proximity to the other BSDs,
ideally next to FreeBSD, as it shares most similarities with FreeBSD.
In this case, FreeBSD is in Tier 1, so mention it after OpenBSD. Sorry
for shifting OmniOS to the right.

As of the naming: We call it "DragonFly" and only sometimes "DragonFly
BSD" when talking to non-DragonFly people.

mneumann commented:
@kit-ty-kate @gasche I've added an entry to Changes. Not sure if this is the right way of doing it, and if I got Kate's name correct :). Also added DragonFly to the list of Tier 2.

nojb reviewed (approved):
LGTM
